cmd/provision/main.go

```diff
@@ -186,30 +193,28 @@ func main() {
 	klog.Info("✅ Device provisioned!")
 }
 
+// Firmware represents a single firmware image to be be installed on a device.
+type fw struct {
+	// Bundle is the firmware bundle to be installed
+	Bundle firmware.Bundle
+	// Block is the location on MMC where the firmware should be installed.
+	Block int64
+	// ConfigBlock is an optional location on the MMC where the loader config block should be
+	// stored. This is only currently used for the bootloader firmware due to location constraints.
+	ConfigBlock int64
+}
+
 // firmwares respresents the collection of firmware and related artefacts which must be
 // flashed onto the device.
 type firmwares struct {
 	// Bootloader holds the regular bootloader firmware bundle.
-	Bootloader firmware.Bundle
-	// BootloaderBlock is the location on MMC where the bootloader should be written.
-	BootloaderBlock int64
-	// BootloaderConfigBlock is the location on the MMC where the bootloader config GOB
-	// should be written. Note that this is different to the other firmware images which
-	// have their config GOBs prepended before writing; we cannot do that with the
-	// bootloader since the mask ROM expects the IMX image to be available at a
-	// particular location very close to the start of the MMC.
-	BootloaderConfigBlock int64
+	Bootloader *fw
```

Comment:
It seems weird to have this as public variable with a private type? Does `fw` not need uppercasing? Or the fields lowercasing?